nltk/translate/lepor.py

```diff
@@ -0,0 +1,253 @@
+                    if distance > min_distance:
+                        min_distance = distance
+                        min_index = ref_index
+                alignments.append(min_index)
+
+    alignments = [a + 1 for a in alignments if a != -1]
+    match_count = len(alignments)
+    npd_list = []
+
+    for ind, a in enumerate(alignments):
+        npd_list.append(abs(((ind + 1) / len(hypothesis_words)) - (a / len(ref_words))))
+    npd = sum(npd_list) / len(hypothesis_words)
+
+    return math.exp(-npd), match_count
+
+
+def harmonic(
+    match_count: int,
+    reference_length: int,
+    hypothesis_length: int,
+    alpha: float,
+    beta: float,
+) -> float:
+    """
+    Function will calculate the precision and recall of matched words and calculate a final score on wighting
+    using alpha and beta parameters.
+
+    :param match_count: number of words in hypothesis aligned with reference.
+    :type match_count: int
+    :param reference_length: length of the reference sentence
+    :type reference_length: int
+    :param hypothesis_length: length of the hypothesis sentence
+    :type hypothesis_length: int
+    :param alpha: a parameter to set weight fot recall.
+    :type alpha: float
+    :param beta: a parameter to set weight fot precision.
+    :type beta: float
+
+    :return: harmonic mean.
+    :rtype: float
+    """
+
+    epsilon = sys.float_info.epsilon
+
+    precision = match_count / hypothesis_length
+    recall = match_count / reference_length
+
+    harmonic_score = (alpha + beta) / (
+        (alpha / (recall + epsilon)) + (beta / (precision + epsilon))
+    )
+
+    return harmonic_score
+
+
+def sentence_lepor(
+    reference: str, hypothesis: str, alpha: float = 1.0, beta: float = 1.0
+) -> float:
+    """
+    Calculate LEPOR score a sentence from Han, A. L.-F. (2017).
+    LEPOR: An Augmented Machine Translation Evaluation Metric. https://arxiv.org/abs/1703.08748v2
+
+    >>> hypothesis = 'a bird is on a stone.'
+
+    >>> reference = 'a bird behind the stone.'
+
+    >>> sentence_lepor(reference, hypothesis)
+    0.7824248013113159
+
+    :param reference: reference sentence
+    :type reference: list(list(str))
+    :param hypothesis: a hypothesis sentence
+    :type hypothesis: list(str)
+    :param alpha: a parameter to set weight fot recall.
+    :type alpha: float
+    :param beta: a parameter to set weight fot precision.
+    :type beta: float
+
+    :return: The Lepor score. Returns a float value of lepor score for a sentence
+    :rtype: float
+
+    """
+
+    if len(reference) == 0 or len(hypothesis) == 0:
+        raise ValueError("One of the sentence is empty. Exit.")
+
+    # Regex to remove all the unnecessary punctuation from the text.
+    reference = nltk.word_tokenize(reference)
+    hypothesis = nltk.word_tokenize(hypothesis)
+
+    # reference = re.findall(r"[\w']+|[.,!?;]", reference)
+    # hypothesis = re.findall(r"[\w']+|[.,!?;]", hypothesis)
```

Comment:
Since this is in NLTK, I'll think `nltk.word_tokenize()` would be expected, but you can try adding simple vs nltk tokenize as a parameter:


```
from nltk.tokenize import word_tokenize

def sentence_lepor(
    reference: str, hypothesis: str, alpha: float = 1.0, beta: float = 1.0, 
    use_nltk_tokenize=True):
    ....
    if use_nltk_tokenize: 
        reference = word_tokenize(reference)
        hypothesis = word_tokenize(hypothesis)
    else:
        reference = re.findall(r"[\w']+|[.,!?;]", reference)
        hypothesis = re.findall(r"[\w']+|[.,!?;]", hypothesis)
```

Reply:
Yes, that's better, Perhaps we can also provide an option to pass a callable tokenizer(if not passed, we can use the one already in nltk).